services/catalog/service/src/graphql/plugins/asset-localization-plugins.ts

```diff
@@ -0,0 +1,113 @@
+import { optional } from '@axinom/mosaic-db-common';
+import {
+  getMappedError,
+  isNullOrWhitespace,
+} from '@axinom/mosaic-service-common';
+import { Inflection, Plugin } from 'graphile-build';
+import {
+  gql,
+  makeExtendSchemaPlugin,
+  makePluginByCombiningPlugins,
+} from 'graphile-utils';
+
+import { makeJSONPgSmartTagsPlugin } from 'graphile-utils';
+import { selectOne } from 'zapatos/db';
+import { Table } from 'zapatos/schema';
+
+const AssetLocalizationSmartTagsPlugin = (assetName: string): Plugin =>
+  makeJSONPgSmartTagsPlugin({
+    version: 1,
+    config: {
+      class: {
+        [`${assetName}_localizations`]: {
+          tags: {
+            omit: true,
+          },
+          attribute: {
+            [`${assetName}_id`]: {
+              tags: {
+                omit: true,
+              },
+            },
+            locale: {
+              tags: {
+                omit: true,
+              },
+            },
+            id: {
+              tags: {
+                omit: true,
+              },
+            },
+            is_default_locale: {
+              tags: {
+                omit: true,
+              },
+            },
+          },
+        },
+      },
+    },
+  });
+
+const AssetLocalizationExtendPlugin = (
+  assetName: string,
+  localizationTable: Table,
+): Plugin => {
+  return makeExtendSchemaPlugin((build) => {
+    const inflection: Inflection = build.inflection;
+    const capitalizedAssetName = inflection.upperCamelCase(assetName);
+    return {
+      typeDefs: gql`
+        extend type ${capitalizedAssetName} {
+          """
+          The localized ${assetName} metadata.
+          """
+          localization(
+            """
+            The localization is returned for the specified locale. If not defined, the default locale will be used instead.
+            """
+            locale: String
+          ): ${capitalizedAssetName}Localization! @requires(columns: ["id"])
+        }
+      `,
+      resolvers: {
+        [capitalizedAssetName]: {
+          localization: async ({ id }: { id: string }, args, context) => {
+            try {
+              const localization = await selectOne(localizationTable, {
```

Comment:
Also just for reference as we did talk about it: N+1 issue

Reply:
Adjusted. Sources:
- https://www.graphile.org/postgraphile/make-extend-schema-plugin/#using-the-pgquery-directive-for-non-root-queries-and-better-performance
- https://github.com/graphile/graphile-engine/blob/5211758b7a48191ffd7600f9f5ae572672ffd221/packages/graphile-utils/__tests__/ExtendSchemaPlugin-pg.test.js#L507-L720